sealtk/noaa/gui/Window.cpp

```diff
@@ -1071,6 +1068,111 @@ QModelIndex WindowPrivate::modelIndex(
   return this->trackModelFilter.mapToSource(filterIndex);
 }
 
+// ----------------------------------------------------------------------------
+void WindowPrivate::zoomExtents(sg::ExtentsTypes types) const
+{
+  auto const link = this->ui.actionViewLink->isChecked();
+
+  // Zoom linked views
+  if (link)
+  {
+    // Get extents for linked views
+    auto worldExtents = QRectF{};
+    for (auto* const w : this->allWindows)
+    {
+      if (w->player->hasTransform())
+      {
+        auto const& windowExtents = w->player->extents(types);
+        if (windowExtents.isValid())
+        {
+          if (worldExtents.isValid())
+          {
+            worldExtents = worldExtents.united(windowExtents);
+          }
+          else
+          {
+            worldExtents = windowExtents;
+          }
+        }
+      }
+    }
+
+    // Apply zoom to linked views
+    if (worldExtents.isValid())
+    {
+      auto* const player = this->eoWindow.player;
+
+      auto const worldSize = QSizeF{player->homographyImageSize()};
+      auto const worldRect = QRectF{{0, 0}, worldSize};
+      player->setCenter(worldExtents.center() - worldRect.center());
+
+      if (types != sg::ImageExtents)
+      {
+        auto const adjust =
+          qMax(10.0, 0.1 * qMax(worldExtents.width(), worldExtents.height()));
+        worldExtents.adjust(-adjust, -adjust, adjust, adjust);
+      }
+
+      auto zoom = qInf();
+      for (auto* const w : this->allWindows)
+      {
+        if (w->player->hasTransform())
+        {
+          auto const viewSize = QSizeF{w->player->size()};
+          auto const zw = viewSize.width() / worldExtents.width();
+          auto const zh = viewSize.height() / worldExtents.height();
+          zoom = qMin(zoom, qMin(zw, zh));
+        }
+      }
+      player->setZoom(static_cast<float>(zoom));
+    }
+  }
+
+  // Zoom non-linked views
+  for (auto* const w : this->allWindows)
+  {
+    if (!link || !w->player->hasTransform())
+    {
+      auto windowExtents = w->player->extents(types);
+      if (windowExtents.isValid())
+      {
+        auto const windowSize = QSizeF{w->player->effectiveImageSize()};
+        auto const windowRect = QRectF{{0, 0}, windowSize};
+        w->player->setCenter(windowExtents.center() - windowRect.center());
+
+        if (types != sg::ImageExtents)
+        {
+          auto const adjust =
+            qMax(10.0, 0.1 * qMax(windowExtents.width(),
+                                  windowExtents.height()));
```

Comment:
Perhaps slightly confusing what the 10.0 represents and the overall goal. Is this a cap on zooming speed?

Reply:
No. As above, this is an adjustment (padding) that is applied to the extents prior to fitting them in the view, which is 10 units¹, 10% of the image width, or 10% of the image height, whichever is largest.

(Due to possible homographies, I believe "unit" is effectively "EO image pixel"... as opposed to, say, image pixels in the view being zoomed. Definitely not *screen* pixels.)